pilot/docker/Dockerfile.proxy

```diff
@@ -10,9 +10,10 @@ ADD pilot-agent /usr/local/bin/pilot-agent
 # istio-proxy" will not match connections from those processes anymore.
 # Instead, rely on the process's effective gid being istio-proxy and create a
 # "-m owner --gid-owner istio-proxy" iptables rule in istio-iptables.sh.
-RUN \
-chgrp 1337 /usr/local/bin/envoy /usr/local/bin/pilot-agent && \
-chmod 2755 /usr/local/bin/envoy /usr/local/bin/pilot-agent
+# TODO: disabling due to https://github.com/istio/istio/issues/5745
+# RUN \
+# chgrp 1337 /usr/local/bin/envoy /usr/local/bin/pilot-agent && \
+# chmod 2755 /usr/local/bin/envoy /usr/local/bin/pilot-agent
```

Comment:
it's odd because the setgid is for the 1337 group and not for root

this could be achieved by creating the user 1337 and the group 1337 and assigning the user to it (instead of setgid)

Reply:
At least it should be equivalent from the point of view of the running binary, though I would need to see the envoy/iptable sides as I'm not sure why we need anything to do with group

@rlenglet who I believe set this up would be able to comment/answer more accurately

Reply:
K8s will always set the primary gid to 1. So this chgrp + the setgid bit are necessary for TPROXY.

Reply:
yes, the running gid is not 1337, it's 0. so setgid happens somewhere and this is what causes the problem.

Reply:
why do we need the group to be special?